### DISCUSSION TITLE
Suggestions for Batch processing (loading PDL, store actions ...)

### DISCUSSION THREAD

**Discussion by hi5** [tannerhelland/PhotoDemon#615] [Ideas]
### 1. Loading PDL files directly

At the moment you can save/load the list of files from to a PDL file (see format example) when you start the Batch process operation.  
But it doesn't seem possible to open the `filelist.pdl` file directly in PD.  
So my suggestion would be for PD to recognize PDL as a valid extention of a file it can open and then open the Batch processing dialog with the files read from the PDL file.

### 2. Store actions / conversions (in PDL files)

It would be nice if you could also export the actions/conversions you've set in the batch processing.

So say you want to convert all files to JPG on a regular basis, storing this "convert to JPG with these settings" would be nice.

If that would be part of the PDL file format (xml?) you could prepare a PDL file yourself and then have PD execute the conversion by simply calling `photodemon.exe filelist.pdl`

*Format example*

The format for PDL files is easy to recreate using a "third party" solution (batch, scripting, other tools).

Format: start with `<PHOTODEMON BATCH CONVERSION LIST>`, next line has number of files, followed by the list of files, and file ends with `<END OF LIST>`

Example:
```
<PHOTODEMON BATCH CONVERSION LIST>
3
C:\path-to-file\file1.svg
C:\path-to-file\file2.svg
C:\path-to-file\file3.svg
<END OF LIST>
```

**Comment by tannerhelland** [comment]
Thank you for this intriguing suggestion, @hi5!  I have been mulling it over and would like to tackle it soon-ish.

The hardest part about saving stuff to a unique file format is making sure the format is (mostly) future-proof, so I won't break it when I inevitably make changes in the future.  I've started jotting down some notes to that end and will soon tackle "batch conversion settings" as its own file type.  Then, I'll work on getting that and "batch file lists" added to the main "open" dialog as another file type (a great idea, one I'm sad I didn't think of myself).

I'll report back when I have something to show.  Thanks!
